### PR TITLE
Typedoc

### PR DESCRIPTION
**Pull request by amcaplan**
### WHY are these changes introduced?

Implements https://github.com/Shopify/internal-cli-foundations/issues/565
Makes it easier to develop on cli-kit.

Screenshot 2023-02-14 at 15 19 30

### WHAT is this pull request doing?

Adds 2 commands:

1. `docs:generate` to generate typedoc docs for `cli-kit`
2. `docs:open` to view the docs locally

We can use `docs:generate` on releases to create a docs microsite.

Note we can't just use `docs` because `pnpm docs` is already a command.

### How to test your changes?

Run the commands! Inside `packages/cli-kit`: `pnpm docs:open` and they'll be viewable in your browser.

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by Arkham** [comment]
Hey, this looks great! do you think it would make sense to add a step in the release where to push to github pages or netlify? could be done in a separate PR of course :)

File: packages/cli-kit/project.json [hunk 1]
```diff
@@ -21,6 +21,22 @@
           "cwd": "packages/cli-kit"
         }
       },
+      "docs:generate": {
+        "executor": "nx:run-commands",
+        "outputs": ["docs"],
+        "options": {
+          "command": "pnpm typedoc --out docs $(find ./src/public/**/{*.ts,*.tsx} | grep -v .test.ts)",
```
Comment (Arkham) [inline packages/cli-kit/project.json:28]: I'm not sure this syntax works across all shells: can you try if this command does the same thing? `find ./src/public -name "*.test.ts" | xargs pnpm typedoc --out docs`

Comment (gonzaloriestra) [reply packages/cli-kit/project.json:28]: Not from Shopify, but we want the community to be able to contribute

**Review by gonzaloriestra — APPROVED** [review]
The generated docs look great! 👌

**Pull request merged**